pkg/scheduler/objects/preemption_queue_test.go

```diff
@@ -123,113 +123,185 @@ func TestGetPreemptableResource(t *testing.T) {
+		childQ2.allocatedResource = smallestRes
+		rootRemaining, pRemaining, cRemaining1, cRemaining2 = setAndGetRemainingGuaranteed(rootQ, parentQ, childQ1, childQ2, queueRes, tt.askQueue)
+		assert.Assert(t, resources.IsZero(rootRemaining), "guaranteed set, used completely. usage also has extra resource types. However, no remaining")
+		assert.Assert(t, resources.IsZero(pRemaining), "guaranteed set, used completely. usage also has extra resource types. However, no remaining")
+		assert.Assert(t, resources.Equals(cRemaining1, resources.Multiply(childRes, -1)), "guaranteed set, used double than guaranteed. so remaining should be in -ve")
+		assert.Assert(t, resources.IsZero(cRemaining2), "guaranteed set, but no usage. Still, no remaining in guaranteed because of its parent queue")
+
+		// clean start for the snapshot: all set guaranteed
+		// add usage for root + parent: use exactly guaranteed for one resource and over guaranteed for another resource at parent level
+		// add usage to child1: use double than guaranteed
+		// add usage to child2: use lesser than guaranteed.
+		// child2 remaining behaviour changes based on the ask queue.
+		// When ask queue is child2, its own values has higher precedence over the parent or ancestor for common resource types.
+		// for extra resources available in parent or ancestor, it can simply inherit.
+		// When ask queue is child3 (diverged from very earlier branch, not sharing any common queue path), its remaining is min permissive of its own values and parent or ancestor values.
+		childQ2.allocatedResource = resources.NewResourceFromMap(map[string]resources.Quantity{"first": 2})
+		rootRemaining, pRemaining, cRemaining1, cRemaining2 = setAndGetRemainingGuaranteed(rootQ, parentQ, childQ1, childQ2, map[string]*resources.Resource{rootQ.QueuePath: resources.Multiply(smallestRes, 2), parentQ.QueuePath: resources.Add(smallestRes, resources.Multiply(childRes, 1)), childQ1.QueuePath: childRes, childQ2.QueuePath: smallestRes}, tt.askQueue)
+		assert.Assert(t, resources.IsZero(rootRemaining), "guaranteed set, used completely. usage also has extra resource types. However, no remaining")
+		assert.Assert(t, resources.DeepEquals(pRemaining, resources.Add(expectedSmallestRes1, resources.Multiply(childRes, -1))), "guaranteed set, one resource type used completely. usage also has another resource types which is used bit more. remaining should have zero for one resource type and -ve for another")
+		assert.Assert(t, resources.DeepEquals(cRemaining1, resources.Add(expectedSmallestRes1, resources.Multiply(childRes, -1))), "guaranteed set, used double than guaranteed. so remaining should be in -ve")
+		assert.Assert(t, resources.DeepEquals(cRemaining2, tt.childQ2Remaining1), "guaranteed set, used bit lesser. parent's usage also has extra resource types. remaining should be based on ask queue")
+	}
+}
+
+func setup(t *testing.T) (*Queue, *Queue, *Queue, *Queue, *Queue) {
```

Comment:
Normally I'm not in favor of named return values, but when there's 5 of the same type... I think it's warranted.